src/engine/SpatialJoinAlgorithms.cpp

```diff
@@ -50,60 +50,142 @@ std::string SpatialJoinAlgorithms::betweenQuotes(
+      // one point and one area
+      AnyGeometry geom1 = geo1.geometry.value();
+      AnyGeometry geom2 = geo2.geometry.value();
+      if (geo1.geoPoint.has_value()) {
+        geom1 = convertGeoPointToPoint(geo1.geoPoint.value());
+      } else if (geo2.geoPoint.has_value()) {
+        geom2 = convertGeoPointToPoint(geo2.geoPoint.value());
+      }
+      return Id::makeFromDouble(convertDegreesToMeters(geom1, geom2));
+    }
+  }
+}
+
 // ____________________________________________________________________________
 Id SpatialJoinAlgorithms::computeDist(const IdTable* idTableLeft,
                                       const IdTable* idTableRight,
                                       size_t rowLeft, size_t rowRight,
                                       ColumnIndex leftPointCol,
                                       ColumnIndex rightPointCol) const {
-  auto getAreaString = [&](const IdTable* idtable, size_t row, size_t col) {
-    return betweenQuotes(ExportQueryExecutionTrees::idToStringAndType(
-                             qec_->getIndex(), idtable->at(row, col), {})
-                             .value()
-                             .first);
-  };
-
   auto getAreaOrPointGeometry =
       [&](const IdTable* idtable, size_t row, size_t col,
           std::optional<GeoPoint> point) -> std::optional<AnyGeometry> {
     AnyGeometry geometry;
-    try {
-      if (!point) {
-        boost::geometry::read_wkt(getAreaString(idtable, row, col), geometry);
-      } else {
-        geometry = Point(point.value().getLng(), point.value().getLat());
+
+    if (!point) {
+      if (!getAnyGeometry(idtable, row, col, geometry)) {
+        // nothing to do. When parsing a point or an area fails, a warning
+        // message gets printed at another place and the point/area just gets
+        // skipped
+        return std::nullopt;
       }
-    } catch (...) {
-      return std::nullopt;
+    } else {
+      geometry = convertGeoPointToPoint(point.value());
     }
+
     return geometry;
   };
 
   // for now we need to get the data from the disk, but in the future, this
   // information will be stored in an ID, just like GeoPoint
   auto getAreaPoint = [&](const IdTable* idtable, size_t row,
                           size_t col) -> std::optional<GeoPoint> {
-    std::string areastring = getAreaString(idtable, row, col);
-
-    Box areaBox;
-    try {
-      areaBox = calculateBoundingBoxOfArea(areastring);
-    } catch (...) {
+    AnyGeometry geometry;
+    if (!getAnyGeometry(idtable, row, col, geometry)) {
       // nothing to do. When parsing a point or an area fails, a warning message
       // gets printed at another place and the point/area just gets skipped
       return std::nullopt;
     }
+
+    Box areaBox = boost::apply_visitor(BoundingBoxVisitor(), geometry);
+
     Point p = calculateMidpointOfBox(areaBox);
     return GeoPoint(p.get<1>(), p.get<0>());
   };
 
+  // if the geometries have not been read already, get them first
```

Comment:
The following code is duplicated to something I have already read above if I am not mistaken.